File: day03/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Println("Error while reading file")
		panic(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	var sumPriorities int
	for scanner.Scan() {
		line := scanner.Text()
		firstHalf := line[:len(line)/2]
		secondHalf := line[len(line)/2:]
		var item rune
		for _, char := range firstHalf {
			if strings.Contains(secondHalf, string(char)) {
				item = char
				break
			}
		}
		if item >= 97 && item <= 122 {
			item -= 96
		} else {
			item -= 38
		}
		sumPriorities += int(item)
	}
	fmt.Println(sumPriorities)
}

File: day04/part1.py
def solution():
    with open('input.txt', 'r') as file:
        count = 0
        for line in file:
            sec1, sec2 = line.split(',')
            x1, y1 = list(map(int, sec1.split('-')))
            x2, y2 = list(map(int, sec2.split('-')))
            if y1 >= x2 and y1 <= y2:
                    count += 1
                    continue
            if y2 >= x1 and y2 <= y1:
                    count += 1
        print(count)

if __name__ == '__main__':
    solution()
